### ISSUE TITLE
Hidden Ground Items Resetting

### ISSUE THREAD

**Issue by kittyk4y** [runelite/runelite#8457]
**Describe the bug**
When removing all items from the "Hidden" Ground Items list, after quitting and restarting the client, they are back.

**To Reproduce**
Go to Ground Items
Erase everything in "Hidden"
Close RuneLite
Open RuneLite
Check Ground Items

**Expected behavior**
Field should stay cleared after clearing.

**Environment (please complete the following information):**
 - OS: Windows 10
 - RuneLite version: 1.5.18
 - Launcher version: Unknown

**Comment by kittyk4y** [comment]
FYI, I tested putting "Iron Daggers" in as a junk item, closed and restarted the client, and it saved it. For some reason, it is not saving the field when blank.

**Comment by raiyni** [comment]
It resets to the default value when blank.

**Comment by kittyk4y** [comment]
Which should be treated as a big, as all the other similar fields don’t act this way, and there is no reason for it to.

**Comment by kittyk4y** [comment]
I’d also like to point out this wasn’t an issue before the last few releases.

**Comment by raiyni** [comment]
Whatever release was 8/28 probably because that is when the default was added.

**Comment by LeMorrow** [comment]
For anyone that is looking for a workaround for not hiding any items, just input anything in "Hidden Items". For example, I just put `Nothing` and it works fine.

![image](https://user-images.githubusercontent.com/35617441/70622298-eb900380-1c1b-11ea-80a6-078e812ea065.png)

**Comment by heyitsmepope** [comment]
> For anyone that is looking for a workaround for not hiding any items, just input anything in "Hidden Items". For example, I just put `Nothing` and it works fine.
> 
> ![image](https://user-images.githubusercontent.com/35617441/70622298-eb900380-1c1b-11ea-80a6-078e812ea065.png)

As of September 1, 2023, placing the text, "Nothing" in the Hidden Items field no longer works. It acts the same as leaving the field blank. Closing or resetting the Runelite app after using this method now results in example:

**Hidden Items**
>Nothing,Green egg,Red egg,Blue egg,Hammer,Logs,Yellow egg,Crackers,Tofu,Worms

**Comment by Adam-** [comment]
> > For anyone that is looking for a workaround for not hiding any items, just input anything in "Hidden Items". For example, I just put `Nothing` and it works fine.
> > ![image](https://user-images.githubusercontent.com/35617441/70622298-eb900380-1c1b-11ea-80a6-078e812ea065.png)
> 
> As of September 1, 2023, placing the text, "Nothing" in the Hidden Items field no longer works. It acts the same as leaving the field blank. Closing or resetting the Runelite app after using this method now results in example:
> 
> **Hidden Items**
> 
> > Nothing,Green egg,Red egg,Blue egg,Hammer,Logs,Yellow egg,Crackers,Tofu,Worms

This is incorrect. Adding "nothing", or anything else, prevents the field from being reset. The default items are also `Vial, Ashes, Coins, Bones, Bucket, Jug, Seaweed`. 

`Green egg,Red egg,Blue egg,Hammer,Logs,Yellow egg,Crackers,Tofu,Worms` are specific to barbarian assault. Maybe you installed a barbarian assault related plugin which is overwriting your hidden items. You should uninstall and stop using such plugins.

**Comment by heyitsmepope** [comment]
> > > For anyone that is looking for a workaround for not hiding any items, just input anything in "Hidden Items". For example, I just put `Nothing` and it works fine.
> > > ![image](https://user-images.githubusercontent.com/35617441/70622298-eb900380-1c1b-11ea-80a6-078e812ea065.png)
> > 
> > 
> > As of September 1, 2023, placing the text, "Nothing" in the Hidden Items field no longer works. It acts the same as leaving the field blank. Closing or resetting the Runelite app after using this method now results in example:
> > **Hidden Items**
> > > Nothing,Green egg,Red egg,Blue egg,Hammer,Logs,Yellow egg,Crackers,Tofu,Worms
> 
> This is incorrect. Adding "nothing", or anything else, prevents the field from being reset. The default items are also `Vial, Ashes, Coins, Bones, Bucket, Jug, Seaweed`.
> 
> `Green egg,Red egg,Blue egg,Hammer,Logs,Yellow egg,Crackers,Tofu,Worms` are specific to barbarian assault. Maybe you installed a barbarian assault related plugin which is overwriting your hidden items. You should uninstall and stop using such plugins.

You are correct. Thank you for that information. The additional Barbarian Assault plugins were indeed causing the issue.

**Comment by ErroneousFatality** [comment]
My friends, could we not resolve this 4 year old bug?

**Comment by CheckYourFax** [comment]
I think the simplest solution to this is to make a toggle for Hidden items (E.G. Hidden items enabled [ ]) and greying out the Hidden items area if that toggle is not on.

**Comment by blitzduck** [comment]
The same problem exists with resetting "Highlighted Items", it always reverts to a previous state, I'm functionally unable to make any changes to it. 

The only solution that persists is creating a brand new Runelite profile (requires reinstalling desired plugins).  

Is there a config file we can edit in Notepad or something?

**Comment by JZomDev** [comment]
> The same problem exists with resetting "Highlighted Items", it always reverts to a previous state, I'm functionally unable to make any changes to it.
> 
> The only solution that persists is creating a brand new Runelite profile (requires reinstalling desired plugins).
> 
> Is there a config file we can edit in Notepad or something?

This is an issue tracker is not a support forum. Please [join the discord](https://runelite.net/discord) if you wish to receive further support.

**Comment by Felanbird** [comment]
> it always reverts to a previous state, I'm functionally unable to make any changes to it.

That's not the same problem, this issue is about how if the input box is empty, the client will revert to the default items, not just "be empty"

The issue you have is very likely a permission issue, the client is no longer able to update its file, which we can give you the steps to fix in the discord

**Comment by Tunatroll** [comment]
> My friends, could we not resolve this 4 year old bug?

7+~ year old bug

**Comment by blitzduck** [comment]
For anyone still experiencing this issue:

Check for the "Highlight Stackables" plugin; it was what caused the issues for me. I uninstalled it and the issue hasn't reoccurred since.

**Comment by Felanbird** [comment]
> For anyone still experiencing this issue:
> 
> Check for the "Highlight Stackables" plugin; it was what caused the issues for me. I uninstalled it and the issue hasn't reoccurred since.

I believe the `Ba Minigame` plugin can also cause this.
